pages/admin/programs/index.tsx

```diff
@@ -0,0 +1,183 @@
+import {
+    Box,
+    Flex,
+    HStack,
+    Link,
+    useColorModeValue,
+    Text,
+    Button,
+    Divider,
+    InputGroup,
+    InputLeftElement,
+    Input,
+    Grid,
+    GridItem,
+    Spacer,
+} from "@chakra-ui/react";
+import Wrapper from "@components/AdminWrapper";
+import React from "react";
+import { ReactNode } from "react";
+import colourTheme from "@styles/colours";
+import { SmallAddIcon } from "@chakra-ui/icons";
+import { SearchIcon } from "@chakra-ui/icons";
+import { BrowseProgramCard } from "@components/BrowseProgramCard";
+import { locale } from "@prisma/client";
+import usePrograms from "@utils/hooks/usePrograms";
+import { useRouter } from "next/router";
+import { Loading } from "@components/Loading";
+import { GetServerSideProps } from "next"; // Get server side props
+import { getSession } from "next-auth/client";
+import { serverSideTranslations } from "next-i18next/serverSideTranslations";
+import { AdminEmptyState } from "@components/admin/AdminEmptyState";
+import { useState } from "react";
+
+type BrowseProgramsProps = {
+    session: Record<string, unknown>;
+};
+
+const Links = [
+    { name: "Add Program", url: "/admin/program/create" },
+    { name: "Add Class", url: "/admin/class/create" },
+];
+
+const NavLink = ({ href, children }: { href: string; children: ReactNode }) => (
+    <Link href={href}>
+        <HStack>
+            <Button
+                fontSize="16px"
+                backgroundColor="white"
+                fontWeight="400"
+                px={"16px"}
+                py={"8px"}
+                border-radius="6px"
+                border="1px solid #0C53A0"
+            >
+                <SmallAddIcon mr="11px" />
+                {children}
+            </Button>
+        </HStack>
+    </Link>
+);
+
+export const BrowsePrograms: React.FC<BrowseProgramsProps> = (props) => {
+    const router = useRouter();
+    const [searchTerm, setSearchTerm] = useState("");
+
+    const {
+        programs: programCardInfos,
+        isLoading,
+        error,
+    } = usePrograms(router.locale as locale);
+
+    if (isLoading) {
+        return <Loading />;
+    } else if (error) {
+        return <Box>An Error has occurred</Box>;
+    }
+
+    const filteredCards = programCardInfos.filter((val) => {
+        if (searchTerm == "") {
+            return val;
+        } else if (
+            val.name.toLowerCase().includes(searchTerm.toLowerCase()) ||
+            val.description.toLowerCase().includes(searchTerm.toLowerCase())
+        ) {
+            return val;
+        }
+    });
```

Comment:
maybe make iterated name more clear + search tags
```suggestion
    const filteredCards = programCardInfos.filter((prog) => {
        const term = searchTerm.toLowerCase();
        if (term == "") {
            return prog;
        } else if (
            prog.name.toLowerCase().includes(term) ||
            prog.description.toLowerCase().includes(term) ||
            prog.onlineFormat.toLowerCase().includes(term) ||
            prog.tag.toLowerCase().includes(term)
        ) {
            return prog;
        }
    });
```